test/lit/passes/string-lowering-instructions.wast

```diff
@@ -236,7 +236,8 @@
     (if (result stringref)
       (i32.const 0)
       (then
-        (ref.null none) ;; this will turn into noextern
+        (ref.null noextern) ;; The change from stringref to externref does not
+                            ;; cause problems here, nothing needs to change.
```

Comment:
Does this unlock any potential code simplifications in StringLowering.cpp?

Reply:
I wondered this too, but the answer is actually no. While we are doing something much simpler, logically - swapping a type for a supertype, rather than an entirely unrelated type - the actual mechanical change is identical. We just swap all appearances of type X for Y, and don't need to handle any subtyping.